tools/fregrid/fregrid.c

```diff
@@ -848,21 +847,29 @@ int main(int argc, char* argv[])
     if(nscalar == 0 && nvector == 0) {
       if(mpp_pe() == mpp_root_pe()) printf("NOTE from fregrid: no scalar and vector field need to be regridded.\n");
       mpp_end();
-      return 0;   
+      return 0;
     }
-    
+
     if(nscalar > 0) {
       scalar_in  = (Field_config *)malloc(ntiles_in *sizeof(Field_config));
       scalar_out = (Field_config *)malloc(ntiles_out *sizeof(Field_config));
     }
     if(nvector > 0) {
-      mpp_error("fregrid: currently does not support vertical interpolation, contact developer");
-      u_in  = (Field_config *)malloc(ntiles_in *sizeof(Field_config));
-      u_out = (Field_config *)malloc(ntiles_out *sizeof(Field_config));    
-      v_in  = (Field_config *)malloc(ntiles_in *sizeof(Field_config));
-      v_out = (Field_config *)malloc(ntiles_out *sizeof(Field_config));
+      if ((opcode & CONSERVE_ORDER1) || (opcode & CONSERVE_ORDER2  )){
+          mpp_error("fregrid: conservative interpolation of vector fields is not supported. \n"
+          "Use bilinear interpolation or independent (scalar) vector components.\n");
```

Comment:
This is already a big improvement to this error message.  To me it would be a tiny bit clearer if the suggestion were phrased like this:
```suggestion
          "Use bilinear interpolation or regrid the vector components independently as scalars.\n");
```

Reply:
Thanks. Will use suggestion.